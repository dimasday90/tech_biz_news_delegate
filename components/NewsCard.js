import { Card, Container, Row, Col } from "react-bootstrap";

export default function News(props) {
  return (
    <Container fluid style={{ marginTop: 22 }}>
      <Row>
        {props.news.map((article, i) => (
          <Col key={i} xs={12} sm={6} md={3} lg={3}>
            <Card style={{ maxHeight: 920, marginBottom: 21 }}>
              <Card.Img
                variant="top"
                src={article.urlToImage}
                alt="No Image Available"
              />
              <Card.Body>
                <Card.Title>{article.title}</Card.Title>
                <Card.Text style={{ fontSize: 12 }}>
                  Author: {article.author || "-"}
                </Card.Text>
                <Card.Text>{article.description}</Card.Text>
                <Card.Link href={article.url} target="_blank">
                  Details
                </Card.Link>
              </Card.Body>
              <Card.Footer>
                <small className="text-muted">
                  Published Time: {new Date(article.publishedAt).toString()}
                </small>
              </Card.Footer>
            </Card>
          </Col>
        ))}
      </Row>
    </Container>
  );
}
