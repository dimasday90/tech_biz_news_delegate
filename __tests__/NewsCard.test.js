import React from "react";
import { render, cleanup, waitForElement } from "@testing-library/react";
import * as nextRouter from "next/router";
import renderer from "react-test-renderer";

import newsapi from "../api";
import NewsCard from "../components/NewsCard";

afterEach(cleanup);

describe("NewsCard Test", () => {
  test("should render newscard component with data", async () => {
    nextRouter.useRouter = jest.fn();
    nextRouter.useRouter.mockImplementation(() => ({ route: "/tech" }));
    const fetchTechUS = await newsapi.v2.topHeadlines({
      country: "us",
      language: "en",
      category: "technology",
      pageSize: 8
    });
    const oneData = fetchTechUS.articles[0];
    const { getByText } = render(<NewsCard news={fetchTechUS.articles} />);
    expect(getByText(oneData.title)).toBeInTheDocument();
    expect(getByText(oneData.description)).toBeInTheDocument();
  });
});

describe("With Snapshot Testing", () => {
  it("should match newscard coding", async () => {
    nextRouter.useRouter = jest.fn();
    nextRouter.useRouter.mockImplementation(() => ({ route: "/tech" }));
    const fetchTechUS = await newsapi.v2.topHeadlines({
      country: "us",
      language: "en",
      category: "technology",
      pageSize: 8
    });
    const component = renderer.create(<NewsCard news={fetchTechUS.articles} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
