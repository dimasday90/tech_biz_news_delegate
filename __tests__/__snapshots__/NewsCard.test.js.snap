// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing should match newscard coding 1`] = `
<div
  className="container-fluid"
  style={
    Object {
      "marginTop": 22,
    }
  }
>
  <div
    className="row"
  >
    <div
      className="col-lg-3 col-md-3 col-sm-6 col-12"
    >
      <div
        className="card"
        style={
          Object {
            "marginBottom": 21,
            "maxHeight": 920,
          }
        }
      >
        <img
          alt="No Image Available"
          className="card-img-top"
          src="https://i.ytimg.com/vi/Yy62wQ2LF8k/maxresdefault.jpg"
        />
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            How Materia Works in Final Fantasy 7 Remake - IGN
          </div>
          <p
            className="card-text"
            style={
              Object {
                "fontSize": 12,
              }
            }
          >
            Author: 
            -
          </p>
          <p
            className="card-text"
          >
            Just like in the original game, Materia can be equipped to your party's weapons and armor to give your team different spells, buffs, abilities, and perks. He...
          </p>
          <a
            className="card-link"
            href="https://www.youtube.com/watch?v=Yy62wQ2LF8k"
            target="_blank"
          >
            Details
          </a>
        </div>
        <div
          className="card-footer"
        >
          <small
            className="text-muted"
          >
            Published Time: 
            Fri Apr 10 2020 07:00:13 GMT+0700 (Western Indonesia Time)
          </small>
        </div>
      </div>
    </div>
    <div
      className="col-lg-3 col-md-3 col-sm-6 col-12"
    >
      <div
        className="card"
        style={
          Object {
            "marginBottom": 21,
            "maxHeight": 920,
          }
        }
      >
        <img
          alt="No Image Available"
          className="card-img-top"
          src="https://gamespot1.cbsistatic.com/uploads/screen_kubrick/1587/15875866/3653188-cyber.jpg"
        />
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            Next-Gen Versions Of Cyberpunk 2077 Won't Release At PS5/Xbox Series X Launch, Will "Come Later" - GameSpot
          </div>
          <p
            className="card-text"
            style={
              Object {
                "fontSize": 12,
              }
            }
          >
            Author: 
            Jordan Ramée
          </p>
          <p
            className="card-text"
          >
            CD Projekt Red has confirmed that next-gen versions of Cyberpunk 2077 are on the table, but we shouldn't expect to see them with the launch of PlayStation 5 and Xbox Series X.
          </p>
          <a
            className="card-link"
            href="https://www.gamespot.com/articles/next-gen-versions-of-cyberpunk-2077-wont-release-a/1100-6475884/"
            target="_blank"
          >
            Details
          </a>
        </div>
        <div
          className="card-footer"
        >
          <small
            className="text-muted"
          >
            Published Time: 
            Fri Apr 10 2020 06:53:43 GMT+0700 (Western Indonesia Time)
          </small>
        </div>
      </div>
    </div>
    <div
      className="col-lg-3 col-md-3 col-sm-6 col-12"
    >
      <div
        className="card"
        style={
          Object {
            "marginBottom": 21,
            "maxHeight": 920,
          }
        }
      >
        <img
          alt="No Image Available"
          className="card-img-top"
          src="https://thumbor.forbes.com/thumbor/fit-in/1200x0/filters%3Aformat%28jpg%29/https%3A%2F%2Fspecials-images.forbesimg.com%2Fimageserve%2F5e8fa6b4b4aaaa0006ecf647%2F0x0.jpg%3FcropX1%3D667%26cropX2%3D2819%26cropY1%3D85%26cropY2%3D1296"
        />
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            Apple Insider Exposes 2020 iPhone, Looks Too Good To Be True - Forbes
          </div>
          <p
            className="card-text"
            style={
              Object {
                "fontSize": 12,
              }
            }
          >
            Author: 
            Gordon Kelly
          </p>
          <p
            className="card-text"
          >
            Verified internal Apple images reveal a stunning new 2020 iPhone design...
          </p>
          <a
            className="card-link"
            href="https://www.forbes.com/sites/gordonkelly/2020/04/09/apple-2020-iphone-12-design-release-camera-ios-14-widgets-new-iphone-update/"
            target="_blank"
          >
            Details
          </a>
        </div>
        <div
          className="card-footer"
        >
          <small
            className="text-muted"
          >
            Published Time: 
            Fri Apr 10 2020 06:28:07 GMT+0700 (Western Indonesia Time)
          </small>
        </div>
      </div>
    </div>
    <div
      className="col-lg-3 col-md-3 col-sm-6 col-12"
    >
      <div
        className="card"
        style={
          Object {
            "marginBottom": 21,
            "maxHeight": 920,
          }
        }
      >
        <img
          alt="No Image Available"
          className="card-img-top"
          src="https://gamespot1.cbsistatic.com/uploads/screen_kubrick/1179/11799911/3652234-wwarzone.jpg"
        />
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            CoD: Modern Warfare Update Season 3 Patch Notes (Xbox One, PS4, PC) - GameSpot
          </div>
          <p
            className="card-text"
            style={
              Object {
                "fontSize": 12,
              }
            }
          >
            Author: 
            Eddie Makuch
          </p>
          <p
            className="card-text"
          >
            Infinity Ward has released the patch notes for the big PS4, Xbox One, and PC update, and there are some major changes now available.
          </p>
          <a
            className="card-link"
            href="https://www.gamespot.com/articles/cod-modern-warfare-warzone-season-3-update-patch-n/1100-6475780/"
            target="_blank"
          >
            Details
          </a>
        </div>
        <div
          className="card-footer"
        >
          <small
            className="text-muted"
          >
            Published Time: 
            Fri Apr 10 2020 05:58:00 GMT+0700 (Western Indonesia Time)
          </small>
        </div>
      </div>
    </div>
    <div
      className="col-lg-3 col-md-3 col-sm-6 col-12"
    >
      <div
        className="card"
        style={
          Object {
            "marginBottom": 21,
            "maxHeight": 920,
          }
        }
      >
        <img
          alt="No Image Available"
          className="card-img-top"
          src="https://cdn.vox-cdn.com/thumbor/l69xtH5If5Af6I_6UYELVCdXxOQ=/402x0:5177x2500/fit-in/1200x630/cdn.vox-cdn.com/uploads/chorus_asset/file/19887934/9p6JvT3g.png"
        />
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            Facebook is adding a Quiet Mode that silences push notifications on mobile - The Verge
          </div>
          <p
            className="card-text"
            style={
              Object {
                "fontSize": 12,
              }
            }
          >
            Author: 
            Nick Statt
          </p>
          <p
            className="card-text"
          >
            Facebook is rolling out a new Quiet Mode as part of its digital wellness dashboard on mobile. The mode lets you silence both in-app notification badges and system-level badges and push notifications. The mode will also remind you that it’s active when you ope…
          </p>
          <a
            className="card-link"
            href="https://www.theverge.com/2020/4/9/21215638/facebook-quiet-mode-silence-push-mobile-notifications"
            target="_blank"
          >
            Details
          </a>
        </div>
        <div
          className="card-footer"
        >
          <small
            className="text-muted"
          >
            Published Time: 
            Fri Apr 10 2020 05:17:54 GMT+0700 (Western Indonesia Time)
          </small>
        </div>
      </div>
    </div>
    <div
      className="col-lg-3 col-md-3 col-sm-6 col-12"
    >
      <div
        className="card"
        style={
          Object {
            "marginBottom": 21,
            "maxHeight": 920,
          }
        }
      >
        <img
          alt="No Image Available"
          className="card-img-top"
          src="https://cdn.vox-cdn.com/thumbor/wDQj4z8Q8CsX-8niPPmCgu3e-XU=/0x115:1600x953/fit-in/1200x630/cdn.vox-cdn.com/uploads/chorus_asset/file/19709641/motogold.jpeg"
        />
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            Motorola’s $1,499 flipping Razr now comes in gold - The Verge
          </div>
          <p
            className="card-text"
            style={
              Object {
                "fontSize": 12,
              }
            }
          >
            Author: 
            Taylor Lyles
          </p>
          <p
            className="card-text"
          >
            Motorola’s foldable Razr phone now comes in a blush gold color variant. The gold version features a black-and-gold color scheme and costs the same price as the black model, which retails for $1,499.
          </p>
          <a
            className="card-link"
            href="https://www.theverge.com/2020/4/9/21215610/motorola-foldable-razr-flip-phone-gold-color-verizon-price"
            target="_blank"
          >
            Details
          </a>
        </div>
        <div
          className="card-footer"
        >
          <small
            className="text-muted"
          >
            Published Time: 
            Fri Apr 10 2020 05:10:10 GMT+0700 (Western Indonesia Time)
          </small>
        </div>
      </div>
    </div>
    <div
      className="col-lg-3 col-md-3 col-sm-6 col-12"
    >
      <div
        className="card"
        style={
          Object {
            "marginBottom": 21,
            "maxHeight": 920,
          }
        }
      >
        <img
          alt="No Image Available"
          className="card-img-top"
          src="https://9to5google.com/wp-content/uploads/sites/4/2020/04/APK-Insight-Google-Stadia-2-13.jpg?resize=1024,512"
        />
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            Google Stadia 2.13 preps touchscreen gamepad, Android TV, more [APK Insight] - 9to5Google
          </div>
          <p
            className="card-text"
            style={
              Object {
                "fontSize": 12,
              }
            }
          >
            Author: 
            -
          </p>
          <p
            className="card-text"
          >
            We've found that Google has begun working on the ability to play Stadia games with touch controls on your phone, as well as Android TV support.
          </p>
          <a
            className="card-link"
            href="https://9to5google.com/2020/04/09/google-stadia-2-13-touch-controls/"
            target="_blank"
          >
            Details
          </a>
        </div>
        <div
          className="card-footer"
        >
          <small
            className="text-muted"
          >
            Published Time: 
            Fri Apr 10 2020 04:52:09 GMT+0700 (Western Indonesia Time)
          </small>
        </div>
      </div>
    </div>
    <div
      className="col-lg-3 col-md-3 col-sm-6 col-12"
    >
      <div
        className="card"
        style={
          Object {
            "marginBottom": 21,
            "maxHeight": 920,
          }
        }
      >
        <img
          alt="No Image Available"
          className="card-img-top"
          src="https://cdn.vox-cdn.com/thumbor/aoX9E_B4pYDnjY2I1jcBPpxzs8g=/0x0:1920x1005/fit-in/1200x630/cdn.vox-cdn.com/uploads/chorus_asset/file/19339189/vlcsnap_2019_11_01_14h52m04s461.jpg"
        />
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            World of Warcraft: Shadowlands gets a new ‘!’ quest marker - Polygon
          </div>
          <p
            className="card-text"
            style={
              Object {
                "fontSize": 12,
              }
            }
          >
            Author: 
            Ryan Gilliam
          </p>
          <p
            className="card-text"
          >
            After nearly two decades, Blizzard has made a change to the famous World of Warcraft quest icons. In the new Shadowlands expansion, players have several new quality of life improvements on the tired questing interface.
          </p>
          <a
            className="card-link"
            href="https://www.polygon.com/2020/4/9/21215595/world-of-warcraft-shadowlands-alpha-quest-icon-ui-upgrade-map-marker-campaign-side"
            target="_blank"
          >
            Details
          </a>
        </div>
        <div
          className="card-footer"
        >
          <small
            className="text-muted"
          >
            Published Time: 
            Fri Apr 10 2020 04:48:23 GMT+0700 (Western Indonesia Time)
          </small>
        </div>
      </div>
    </div>
  </div>
</div>
`;
